test/coverage/communication/PublisherTest.cpp

```diff
@@ -11,30 +11,93 @@
 
 #include <gtest/gtest.h>
 #include <up-cpp/communication/Publisher.h>
+#include <up-cpp/datamodel/builder/Payload.h>
+#include <up-cpp/datamodel/serializer/UUri.h>
+#include <uprotocol/v1/uri.pb.h>
 
 #include "UTransportMock.h"
+#include "up-cpp/communication/Publisher.h"
+#include "up-cpp/datamodel/builder/UMessage.h"
+#include "up-cpp/transport/UTransport.h"
 
+using namespace uprotocol::communication;
+using namespace uprotocol::datamodel::builder;
+using namespace uprotocol::v1;
+using ::testing::_;
+using ::testing::Return;
 namespace {
+using namespace uprotocol::datamodel::builder;
 
-class TestFixture : public testing::Test {
+class TestPublisher : public testing::Test {
 protected:
 	// Run once per TEST_F.
 	// Used to set up clean environments per test.
-	void SetUp() override {}
+	void SetUp() override {
+		source_.set_authority_name("10.0.0.1");
+		source_.set_ue_id(0x00011101);
+		source_.set_ue_version_major(0xF1);
+		source_.set_resource_id(0x0);
+
+		topic_.set_authority_name("10.0.0.1");
+		topic_.set_ue_id(0x00011101);
+		topic_.set_ue_version_major(0xF8);
+		topic_.set_resource_id(0x8101);
+		transportMock_ =
+		    std::make_shared<uprotocol::test::UTransportMock>(source_);
+
+		format_ = UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+		priority_ = UPriority::UPRIORITY_CS1;
+		ttl_ = std::chrono::milliseconds(1000);
+	}
+
 	void TearDown() override {}
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestPublisher() = default;
+	~TestPublisher() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
+
+	std::shared_ptr<uprotocol::test::UTransportMock> transportMock_;
+	UUri source_;
+	UUri topic_;
+	UPayloadFormat format_;
+	std::optional<UPriority> priority_;
+	std::optional<std::chrono::milliseconds> ttl_;
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+TEST_F(TestPublisher, PublisherSuccess) {
+	std::string testPayloadStr = "test_payload";
+	Publisher publisher(transportMock_, std::move(topic_), format_, priority_,
+	                    ttl_);
+
+	uprotocol::v1::UStatus retval;
+	retval.set_code(uprotocol::v1::UCode::OK);
+	transportMock_->send_status_ = retval;
+
+	Payload testPayload(testPayloadStr, format_);
+	auto status = publisher.publish(std::move(testPayload));
+
+	EXPECT_EQ(status.code(), retval.code());
```

Comment:
Add a check here against `transport_mock_->message_` - it should pass the `isValidPublish()` check in the message validator.